drivers/include/lm75.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_lm75
+ *
+ * @{
+ * @file
+ * @brief       Driver for the LM75 temperature sensor.
+ *
+ * @author      Vitor Batista <vitor.batista@ml-pa.com>
+ *
+ * @}
+ */
+
+#ifndef LM75_H
+#define LM75_H
+
+#include "periph/i2c.h"
+
+/**
+ * @name LM75 return values
+ */
+enum {
+    LM75_SUCCESS = 0,
+    LM75_ERROR_I2C,
+    LM75_ERROR
+};
+
+/* temperature resolution of the LM75 sensors */
+typedef struct {
+    uint16_t    os_res;
+    uint8_t     os_bits;
+    uint16_t    os_mult;
+    uint16_t    temp_res;
+    uint8_t     temp_bits;
+    uint16_t    temp_mult;
+} lm75_temp_resolutions;
+
+/* LM75 configuration struct type */
+typedef struct lm75_params {
+    i2c_t   i2c_bus;
+    uint8_t i2c_addr;
+    uint8_t shutdown_mode;
+    uint8_t tm_mode;
+    uint8_t polarity;
+    uint8_t fault_q;
+    uint8_t conv_rate;
+    uint8_t one_shot_mode;
+    uint8_t device_type;
+    lm75_temp_resolutions lm75_res;
+} lm75_params_t;
+
+/* M75 device descriptor */
+typedef struct lm75 {
+    lm75_params_t	lm75_params;
+} lm75_t;
+
+
+/**
+ * @brief Initialization of the LM75 sensor
+ *
+ * Initializes the sensor according to specific input parameters.
+ *
+ * @param[out] dev        device structure to initialize
+ * @param[in] params      initialization parameters
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on initialization related error
+ */
+int lm75_init(lm75_t *dev, const lm75_params_t *params);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Reads the sensor temperature values from TEMP_REG.
+ *
+ * @param[out] dev                device structure
+ * @param[out] temperature        buffer where temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_temperature(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Sets the values for Overtemperature shutdown(OS) and Hysteresis temperature(HYST).
+ * OS gives the temperature's higher bound and HYST the lower bound
+ *
```

Comment:
What happens in case of an alarm?

Reply:
Depends on whether the device is in comparator or interrupt mode, as well as on the configured fault queue, but basically it's the following:
>The main difference between the two modes is that in OS comparator mode, the OS
output becomes active when Temp has exceeded Tos and reset when Temp has dropped
below Thyst, reading a register or putting the device into shutdown mode does not change
the state of the OS output; while in OS interrupt mode, once it has been activated either
by exceeding Tos or dropping below Thyst, the OS output will remain active indefinitely until
reading a register, then the OS output is reset.

Reply:
But the driver doesn't support configuring this, does it?
It also doesn't configure the OS pin.

Reply:
The driver allows configuring both through the Kconfig file, but none of them are currently used by any functions.